### PR TITLE
Fixed issue 104 and added test. 

### PR DESCRIPTION
**Pull request by grantfar**
Changed method to use ToString instead of casting

### CODE REVIEW

File: src/LitJson/JsonMapper.cs [hunk 1]
```diff
@@ -782,7 +782,7 @@ private static void RegisterBaseImporters ()
             if (obj is IDictionary) {
                 writer.WriteObjectStart ();
                 foreach (DictionaryEntry entry in (IDictionary) obj) {
-                    writer.WritePropertyName ((string) entry.Key);
+                    writer.WritePropertyName (entry.Key.ToString());
```
Comment (devlead) [inline src/LitJson/JsonMapper.cs]: To not get any culture issues when serializing yes, we should use invariant culture, something like 
```csharp
            if (obj is IDictionary dictionary) {
                writer.WriteObjectStart ();
                foreach (DictionaryEntry entry in dictionary) {
                    var propertyName = entry.Key is string key
                        ? key
                        : string.Format(CultureInfo.InvariantCulture, "{0}", entry.Key);

                    writer.WritePropertyName (propertyName);
                    WriteValue (entry.Value, writer, writer_is_private,
                                depth + 1);
                }
                writer.WriteObjectEnd ();

                return;
            }
```

Comment (devlead) [reply src/LitJson/JsonMapper.cs]: For perf it should me `Convert.ToString(entry.Key, CultureInfo.InvariantCulture);` and not `string format`.

**Comment by grantfar** [comment]
I have made the changes you requested

File: src/LitJson/JsonMapper.cs (hunk 1 above)
Comment (grantfar) [reply src/LitJson/JsonMapper.cs]: I have made the changes you requested

**Comment by grantfar** [comment]
Could you please review my changes

**Comment by devlead** [comment]
@grantfar your changes have been merged, thanks for your contribution 👍

**Pull request merged**